tiny_dnn/util/parallel_for.h

```diff
@@ -82,6 +86,31 @@ void parallel_for(int begin, int end, const Func &f, int /*grainsize*/) {
   for (int i = begin; i < end; ++i) f(blocked_range(i, i + 1));
 }
 
+#elif defined(CNN_USE_GCD)
+
+template <typename Func>
+void parallel_for(int begin, int end, const Func &f, int grainsize) {
+  int count     = end - begin;
+  int blockSize = grainsize;
+  if (count < blockSize || blockSize == 0) {
+    blockSize = 1;
+  }
+  int blockCount = (count + blockSize - 1) / blockSize;
+  assert(blockCount > 0);
+
+  dispatch_apply(blockCount, dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0),
+                 ^(size_t block) {
```

Comment:
Not all compilers support the block syntax `^`, so I thought using `__OBJC__` macro to check if Objective C compiler is being used might be good.

Someone may want to use GCD with other compilers, but that's a minority so supporting it can be postponed.